src/common/helpers/data-extract/data-extract.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * @typedef  { 'name' | 'file' | 'text' |' radio' | 'address' | 'checkbox' } AnswerType
+ * @typedef  {{ type: AnswerType, value: string, displayText: string }} AnswerData
+ * @typedef  {{ question: string, questionKey: string, answer: AnswerData }} QuestionAnswerData
+ * @typedef  {{ title: string, sectionKey: string, questionAnswers: QuestionAnswerData[]}} SectionData
+ */
+
+/**
+ * @param {string} questionKey
+ * @param {string} sectionKey
+ * @param {SectionData[]} sections
+ * @returns {QuestionAnswerData|undefined}
+ */
+export const getQuestionFromSections = (questionKey, sectionKey, sections) => {
+  const section = Array.isArray(sections)
+    ? sections?.find((section) => section.sectionKey === sectionKey)
+    : {}
+  const question = Array.isArray(section?.questionAnswers)
+    ? section?.questionAnswers?.find(
+        (question) => question.questionKey === questionKey
+      )
+    : undefined
+  return question
+}
+
+/**
+ * @param {object} payload
```

Comment:
I'd argue it's neater to explicitly provide a type of an `Application` payload sent to submit - and use it here & also in the handler?

Reply:
done